### PR TITLE
Fix Webhook creation.

### PR DESCRIPTION
**Pull request by DominicGunn**
## Changes

Attempting to create webhooks fail unless a name is specified. As per the documentation the only name that is valid is "web", so I haven't opted to allow modification of this but creation seems to fail without it.

### CODE REVIEW

**Comment by DominicGunn** [comment]
I signed it!

**Comment by gmlewis** [comment]
The only documentation I found about this:
https://developer.github.com/v3/repos/hooks/#create-a-hook
says that the `name` field has a default value of `web`.

Do you want to contact GitHub (support@github.com) and ask them to clarify what is going on after describing the situation you discovered? I have found their support team to be incredibly responsive and helpful.

**Comment by radeksimko** [comment]
I have just emailed GitHub support about this, will update here once I get response from them.

**Comment by radeksimko** [comment]
Here's a response I received from helpful GitHub support folk:

----

This field is currently a legacy field, in that we only provide a single option for this: 'web' for webhooks. Previously we had the concept of GitHub Services: https://developer.github.com/changes/2018-10-01-denying-new-github-services/ that allowed you to create hooks for this.

However as mentioned in the above article we have since deprecated and switched off this service. We decided to leave this field in, to prevent issues with current integrations that might still rely on this. In this case it automatically defaults to 'web' if omitted, or if provided, must have the value 'web'.

For instance the following core curl request works:
```
curl -H "Accept: application/vnd.github.v3+json" \
-H "Authorization: token $TOKEN" \
-d "{\"config\":{\"url\":\"$URL\"}}" \
https://api.github.com/repos/mattamorphic/WebhookTest/hooks
```
Where $TOKEN is the OAuth token, and $URL is the resource to send the requests to.

Long term we could choose to add other values to this as we are often iterating on the services that we provide, we could also choose to in the long term officially deprecate and remove this field as part of an ongoing effort to improve our validation rules.

In terms of the PR, https://github.com/google/go-github/pull/1117, the author @DominicGunn mentions that:

> Attempting to create webhooks fail unless a name is specified

As we haven't been able to reproduce this with the above curl I'd be interested to know if the user can reproduce this with a `curl -v`?

TL;DR; This field has a single value, is a legacy field, and has a default value if omitted.

In terms of defining this field in the SDK, I feel like the SDK should emulate the API behaviour:

1. Optional, and if not provided default to 'web'
1. If provided this can only accept 'web'

----

I'm not sure if the SDK has any convention around validation of arguments, which may need to be added, but this PR otherwise looks good to me according to the above response.

File: github/repos_hooks.go [hunk 1]
```diff
@@ -92,6 +92,7 @@ func (h Hook) String() string {
 // information.
 type createHookRequest struct {
 	// Config is required.
+	Name   string                 `json:"name,omitempty"`
```
Comment (gmlewis) [inline github/repos_hooks.go]: Please remove the `,omitempty` since it is not a pointer and with this PR, it will always be defined.

**Comment by gmlewis** [comment]
Note that this PR does not implement the recommendation in the [above comment](https://github.com/google/go-github/pull/1117#issuecomment-467513337) but it sounds to me like the API is not changing, so this solution would solve the issue and is fine with me.

We'll see if @gauntface agrees.

File: github/repos_hooks.go (hunk 1 above)
Comment (DominicGunn) [reply github/repos_hooks.go]: Removed as per request.

**Comment by DominicGunn** [comment]
For some deeper clarity, this is against a enterprise version of GHE, I'm not sure how much difference that would make. 

In terms of allowing this to be definable by the user, I'm not against it but it feels superfluous given that the only accepted value will be "web".

**Comment by gauntface** [comment]
I'm ok with this landing, but it does sound like a bug in GHE if you are getting an error without the name field.

**Comment by radeksimko** [comment]
> In terms of allowing this to be definable by the user, I'm not against it but it feels superfluous given that the only accepted value will be "web".

Having it definable would help us a lot in the Terraform GitHub provider as we [still haven't upgraded](https://github.com/terraform-providers/terraform-provider-github/pull/186) SDK since this field was removed and if the field is put back we can deprecate it more gracefully and easily, rather than just pulling the cord and forcing everyone to change their configurations.

**Comment by sourabh3b** [comment]
Hi All,
Adding my thoughts to the conversation.
I am facing the same issue.
Getting this error, while adding a webhook to a repo using go-github client : `"name" wasn't supplied. []`
 It's due to the field "name" in POST `/repos/:owner/:repo/hooks` API. The documentation at https://developer.github.com/v3/repos/hooks/#create-a-hook says that `name` is not required, whereas enterprise documentation at https://developer.github.com/enterprise/2.14/v3/repos/hooks/#create-a-hook says that both `name` and `config` are required. Hence I am able to add webhook using Postman but not using go-github client.
